charts/ingress-nginx/templates/controller-servicemonitor.yaml

```
@@ -45,6 +45,21 @@ spec:
   {{- range .Values.controller.metrics.serviceMonitor.targetLabels }}
     - {{ . }}
   {{- end }}
+{{- end }}
+{{- if .Values.controller.metrics.serviceMonitor.labelLimit }}
+  labelLimit: {{- range .Values.controller.metrics.serviceMonitor.labelLimit }}
```

I don't think the use if `range` is correct here. Also please do not trim whitespaces between the field name and its value.

```suggestion
  labelLimit: {{ .Values.controller.metrics.serviceMonitor.labelLimit }}
```

The same goes for the other changes.